### PR TITLE
Application recording flow

### PR DESCRIPTION
**Pull request by ceoaliongroo**
WIP: https://github.com/Gizra/public_education/issues/4

### CODE REVIEW

File: app/scripts/controllers/addMarker.js [hunk 1]
```diff
@@ -71,11 +73,18 @@ angular.module('publicEducationApp')
      * Helper function to indicate recording has completed.
      */
     $scope.onRecorded = function() {
-      $scope.setState('completed');
+      ($scope.user === null) ? $scope.setState('credentials') : $scope.setState('upload');
```
Comment (amitaibu) [inline app/scripts/controllers/addMarker.js]: You should probably wrap this logic with `User.gettingUser()`

File: app/scripts/controllers/addMarker.js [hunk 2]
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
...
      * Helper function to indicate recording has completed.
      */
-    $scope.onRecorded = function() {
-      $scope.setState('completed');
+    $scope.onRecorded = function(form) {
+      //$scope.file = $scope.file;
+      User.getUser().then(function(data) {
+        if (data === null) {
+          $scope.setState('credentials')
+        } else {
```
Comment (ceoaliongroo) [inline app/scripts/controllers/addMarker.js]: line break.

File: app/scripts/controllers/addMarker.js [hunk 3]
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
...
-    $scope.$watch('state', function(newVal, oldVal) {
+    /**
+     * Helper function to indicate file was uploaded successfully to the server.
+     */
+    $scope.onRecordUploaded= function() {
+      $scope.setState('completed');
+    }
 
+    $scope.$watch('state', function(newVal, oldVal) {
+      console.log(oldVal, ', ', newVal);
```
Comment (ceoaliongroo) [inline app/scripts/controllers/addMarker.js]: Remove.

File: app/scripts/controllers/addMarker.js [hunk 4]
```diff
@@ -70,18 +69,38 @@ angular.module('publicEducationApp')
...
       if (oldVal === 'completed') {
         return $scope.onComplete();
       }
 
       if (newVal === 'completed') {
 
+      }
+
+      //Upload marker
```
Comment (ceoaliongroo) [inline app/scripts/controllers/addMarker.js]: => `// Upload marker.`

File: app/views/add-marker.html [hunk 5]
```diff
@@ -63,3 +64,46 @@
...
+
+<div class="share-wrapper" ng-show="state == 'share'">
+  <div>
+    {{user.data.name}}
+    <h1>Thanks</h1>
+    <p>
+      Your command has been recorded.
+    </p>
+    <p>
+      now, go tall your friends!
```
Comment (ceoaliongroo) [inline app/views/add-marker.html]: tall => tell.

File: app/views/play-mode.html [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  <div>
+    <div>
+      <a ng-href="#/add-marker">Add marker</a>
+    </div>
+    <div>
+      <a ng-href="#/">Close</a>
+    </div>
+  </div>
+  <div>
+    <img src="http://placehold.it/71x59" alt=""/>
```
Comment (ceoaliongroo) [inline app/views/play-mode.html]: Show a zoomed, no draggable, map instead of the placeholder.

**Comment by amitaibu** [comment]
Closing, as already added a "play-marker" page, based on this code.

**Pull request closed without merging**